### PR TITLE
VMO-6083 Fix NumericBlock validation

### PR DESCRIPTION
**Pull request by object-required**
We tinker with `ajvErrors` to inject custom error messages. This PR adds explicit handling of such cases.

I also went ahead and rearranged checks specific for _NumericBlock_ in `MobilePrimitives_NumericResponseBlockStore.ts` so that we don't have multiple messages for a single field.

### CODE REVIEW

File: src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts [hunk 1]
```diff
@@ -72,70 +109,58 @@ const actions: ActionTree<IEmptyState, IRootState> = {
...
-            message: lang.trans('flow-builder-validation.numeric-block-min-value-must-be-lower-than-implied-max-digits'),
-          } as ErrorObject)
-          validationResults.ajvErrors.push({
-            dataPath: '/config/validation_minimum',
-            message: lang.trans('flow-builder-validation.numeric-block-min-value-must-be-lower-than-implied-max-digits'),
-          } as ErrorObject)
-        }
+      // Both validationMin and validationMax must have no more digits than allowed
+      if (hasMaxDigits) {
+        const maxValue = 10 ** maxDigits - 1
```
Comment (object-required) [inline src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts:144]: I noticed having `maxDigits = 2` the validation won't accept values over `18` which doesn't feel right.

cc: @pascaleli

**Review by object-required — COMMENTED** [review]
I'm uncertain if we'd better to extract the validation logic specific for _Numeric Block_ to a separate file because it's quite complicated.

**Review by bulatgab — APPROVED** [review]

File: src/store/validation/index.ts [hunk 2]
```diff
@@ -97,7 +97,7 @@ export const actions: ActionTree<IValidationState, IRootState> = {
       ...status,
       ajvErrors: getLocalizedAjvErrors(key, status.ajvErrors),
     })
-    if (status.ajvErrors === null) {
+    if (status.ajvErrors === null || status.ajvErrors.length === 0) {
```
Comment (bulatgab) [inline src/store/validation/index.ts:100]: We could alternatively return null from `errors.list()` to match the ajv interface
(and change logic for `validationResults.ajvErrors`)

File: src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts (hunk 1 above)
Comment (safydy-r) [reply src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts:144]: @object-required , great catch here. Thank you very much 🙏

Comment (safydy-r) [reply src/store/flow/block-types/MobilePrimitives_NumericResponseBlockStore.ts:144]: @object-required , great catch here. Thank you very much 🙏

**Pull request merged**